File: July-20-2025/twoSum.js
var nums =[2,7,11,15]
var target = 9;


var twoSum = function(nums, target) {
  let hash = {}
  
  for(let i = 0; i < nums.length; i++){
      var remainder = target - nums[i]

      if(remainder in hash){
          return [hash[remainder], i]
      }
      hash[nums[i]] = i
  }
  return hash
};

console.log(twoSum(nums, target))

File: November-7-2025/twoSum.py
from typing import List

nums = [3,4,5,6]

target = 7

def two_sum(nums: List[int], target: int) -> List[int]:
  hash = {}

  for i in range(0, len(nums)):
    remainder = target - nums[i]

    if remainder in hash: 
      return [hash[remainder], i]
    else:
      hash[nums[i]] = i
  return []


print(two_sum(nums, target))